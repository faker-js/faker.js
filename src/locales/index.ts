/*
 * This file is automatically generated.
 * Run 'pnpm run generate:locales' to update.
 */

import { default as af_ZA } from './af_ZA';
import { default as ar } from './ar';
import { default as az } from './az';
import { default as base } from './base';
import { default as cs_CZ } from './cs_CZ';
import { default as da } from './da';
import { default as de } from './de';
import { default as de_AT } from './de_AT';
import { default as de_CH } from './de_CH';
import { default as dv } from './dv';
import { default as el } from './el';
import { default as en } from './en';
import { default as en_AU } from './en_AU';
import { default as en_AU_ocker } from './en_AU_ocker';
import { default as en_BORK } from './en_BORK';
import { default as en_CA } from './en_CA';
import { default as en_GB } from './en_GB';
import { default as en_GH } from './en_GH';
import { default as en_HK } from './en_HK';
import { default as en_IE } from './en_IE';
import { default as en_IN } from './en_IN';
import { default as en_NG } from './en_NG';
import { default as en_US } from './en_US';
import { default as en_ZA } from './en_ZA';
import { default as eo } from './eo';
import { default as es } from './es';
import { default as es_MX } from './es_MX';
import { default as fa } from './fa';
import { default as fi } from './fi';
import { default as fr } from './fr';
import { default as fr_BE } from './fr_BE';
import { default as fr_CA } from './fr_CA';
import { default as fr_CH } from './fr_CH';
import { default as fr_LU } from './fr_LU';
import { default as fr_SN } from './fr_SN';
import { default as he } from './he';
import { default as hr } from './hr';
import { default as hu } from './hu';
import { default as hy } from './hy';
import { default as id_ID } from './id_ID';
import { default as it } from './it';
import { default as ja } from './ja';
import { default as ka_GE } from './ka_GE';
import { default as ko } from './ko';
import { default as lv } from './lv';
import { default as mk } from './mk';
import { default as nb_NO } from './nb_NO';
import { default as ne } from './ne';
import { default as nl } from './nl';
import { default as nl_BE } from './nl_BE';
import { default as pl } from './pl';
import { default as pt_BR } from './pt_BR';
import { default as pt_PT } from './pt_PT';
import { default as ro } from './ro';
import { default as ro_MD } from './ro_MD';
import { default as ru } from './ru';
import { default as sk } from './sk';
import { default as sr_RS_latin } from './sr_RS_latin';
import { default as sv } from './sv';
import { default as th } from './th';
import { default as tr } from './tr';
import { default as uk } from './uk';
import { default as ur } from './ur';
import { default as uz_UZ_latin } from './uz_UZ_latin';
import { default as vi } from './vi';
import { default as yo_NG } from './yo_NG';
import { default as zh_CN } from './zh_CN';
import { default as zh_TW } from './zh_TW';
import { default as zu_ZA } from './zu_ZA';

export { default as af_ZA } from './af_ZA';
export { default as ar } from './ar';
export { default as az } from './az';
export { default as base } from './base';
export { default as cs_CZ } from './cs_CZ';
export { default as da } from './da';
export { default as de } from './de';
export { default as de_AT } from './de_AT';
export { default as de_CH } from './de_CH';
export { default as dv } from './dv';
export { default as el } from './el';
export { default as en } from './en';
export { default as en_AU } from './en_AU';
export { default as en_AU_ocker } from './en_AU_ocker';
export { default as en_BORK } from './en_BORK';
export { default as en_CA } from './en_CA';
export { default as en_GB } from './en_GB';
export { default as en_GH } from './en_GH';
export { default as en_HK } from './en_HK';
export { default as en_IE } from './en_IE';
export { default as en_IN } from './en_IN';
export { default as en_NG } from './en_NG';
export { default as en_US } from './en_US';
export { default as en_ZA } from './en_ZA';
export { default as eo } from './eo';
export { default as es } from './es';
export { default as es_MX } from './es_MX';
export { default as fa } from './fa';
export { default as fi } from './fi';
export { default as fr } from './fr';
export { default as fr_BE } from './fr_BE';
export { default as fr_CA } from './fr_CA';
export { default as fr_CH } from './fr_CH';
export { default as fr_LU } from './fr_LU';
export { default as fr_SN } from './fr_SN';
export { default as he } from './he';
export { default as hr } from './hr';
export { default as hu } from './hu';
export { default as hy } from './hy';
export { default as id_ID } from './id_ID';
export { default as it } from './it';
export { default as ja } from './ja';
export { default as ka_GE } from './ka_GE';
export { default as ko } from './ko';
export { default as lv } from './lv';
export { default as mk } from './mk';
export { default as nb_NO } from './nb_NO';
export { default as ne } from './ne';
export { default as nl } from './nl';
export { default as nl_BE } from './nl_BE';
export { default as pl } from './pl';
export { default as pt_BR } from './pt_BR';
export { default as pt_PT } from './pt_PT';
export { default as ro } from './ro';
export { default as ro_MD } from './ro_MD';
export { default as ru } from './ru';
export { default as sk } from './sk';
export { default as sr_RS_latin } from './sr_RS_latin';
export { default as sv } from './sv';
export { default as th } from './th';
export { default as tr } from './tr';
export { default as uk } from './uk';
export { default as ur } from './ur';
export { default as uz_UZ_latin } from './uz_UZ_latin';
export { default as vi } from './vi';
export { default as yo_NG } from './yo_NG';
export { default as zh_CN } from './zh_CN';
export { default as zh_TW } from './zh_TW';
export { default as zu_ZA } from './zu_ZA';

export const allLocales = {
  af_ZA,
  ar,
  az,
  base,
  cs_CZ,
  da,
  de,
  de_AT,
  de_CH,
  dv,
  el,
  en,
  en_AU,
  en_AU_ocker,
  en_BORK,
  en_CA,
  en_GB,
  en_GH,
  en_HK,
  en_IE,
  en_IN,
  en_NG,
  en_US,
  en_ZA,
  eo,
  es,
  es_MX,
  fa,
  fi,
  fr,
  fr_BE,
  fr_CA,
  fr_CH,
  fr_LU,
  fr_SN,
  he,
  hr,
  hu,
  hy,
  id_ID,
  it,
  ja,
  ka_GE,
  ko,
  lv,
  mk,
  nb_NO,
  ne,
  nl,
  nl_BE,
  pl,
  pt_BR,
  pt_PT,
  ro,
  ro_MD,
  ru,
  sk,
  sr_RS_latin,
  sv,
  th,
  tr,
  uk,
  ur,
  uz_UZ_latin,
  vi,
  yo_NG,
  zh_CN,
  zh_TW,
  zu_ZA,
} as const;
