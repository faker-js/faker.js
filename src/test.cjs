const { faker } = require('dist/faker.cjs');

console.log(faker.internet.userAgent());
