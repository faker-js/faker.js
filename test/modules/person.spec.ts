import { describe, expect, it } from 'vitest';
import { Sex, faker, fakerAZ, fakerMK, fakerUK } from '../../src';
import { seededTests } from '../support/seeded-runs';
import { times } from './../support/times';

const NON_SEEDED_BASED_RUN = 5;

describe('person', () => {
  seededTests(faker, 'person', (t) => {
    t.itEach(
      'gender',
      'jobTitle',
      'jobDescriptor',
      'jobArea',
      'jobType',
      'bio'
    );

    t.describe('sexType', (t) =>
      t
        .it('noArgs')
        .it('with includeGeneric=true', { includeGeneric: true })
        .it('with includeGeneric=false', { includeGeneric: false })
    );

    t.describeEach(
      'firstName',
      'lastName',
      'middleName',
      'prefix',
      'sex',
      'suffix'
    )((t) => t.it('noArgs').it('with sex', 'male'));

    t.describe('fullName', (t) => {
      t.it('noArgs')
        .it('with firstName', { firstName: 'John' })
        .it('with lastName', { lastName: 'Doe' })
        .it('with sex', { sex: 'female' })
        .it('with all (sex)', {
          firstName: 'John',
          lastName: 'Doe',
          sex: 'female',
        });
    });

    t.it('zodiacSign');
  });

  describe.each(times(NON_SEEDED_BASED_RUN).map(() => faker.seed()))(
    'random seeded tests for seed %i',
    () => {
      describe('firstName()', () => {
        it('should return a random first name', () => {
          const first_name = faker.person.firstName();

          expect(first_name).toBeTypeOf('string');
          expect(first_name.length).toBeGreaterThan(0);
        });

        it('should return a sex-specific first name', () => {
          let name = faker.person.firstName('generic');
          expect(faker.definitions.person.first_name.generic).toContain(name);

          name = faker.person.firstName('female');
          const female_applicable = [
            ...(faker.definitions.person.first_name.generic ?? []),
            ...(faker.definitions.person.first_name.female ?? []),
          ];
          expect(female_applicable).toContain(name);

          name = faker.person.firstName('male');
          const male_applicable = [
            ...(faker.definitions.person.first_name.generic ?? []),
            ...(faker.definitions.person.first_name.male ?? []),
          ];
          expect(male_applicable).toContain(name);
        });

        it('should return a sex-specific first name when no sex-specific first name was defined', () => {
          const name = fakerAZ.person.firstName();
          expect([
            ...(fakerAZ.definitions.person.first_name.female ?? []),
            ...(fakerAZ.definitions.person.first_name.male ?? []),
          ]).toContain(name);
        });
      });

      describe('lastName()', () => {
        it('should return a random last name', () => {
          const last_name = faker.person.lastName();

          expect(last_name).toBeTypeOf('string');
          expect(last_name.length).toBeGreaterThan(0);
        });

        it('should return a sex-specific last name', () => {
          let name = fakerAZ.person.lastName('female');
          expect(fakerAZ.definitions.person.last_name.female).toContain(name);

          name = fakerAZ.person.lastName('male');
          expect(fakerAZ.definitions.person.last_name.male).toContain(name);
        });
      });

      describe('middleName()', () => {
        it('should return a random middle name', () => {
          const middle_name = faker.person.middleName();

          expect(middle_name).toBeTypeOf('string');
          expect(middle_name.length).toBeGreaterThan(0);
        });

        it('should return a middle name when passed en locale', () => {
          let name = faker.person.middleName();
          const allApplicable = [
            ...(faker.definitions.person.middle_name.generic ?? []),
            ...(faker.definitions.person.middle_name.female ?? []),
            ...(faker.definitions.person.middle_name.male ?? []),
          ];
          expect(allApplicable).toContain(name);

          name = faker.person.middleName('generic');
          expect(faker.definitions.person.middle_name.generic).toContain(name);

          name = faker.person.middleName('female');
          const female_applicable = [
            ...(faker.definitions.person.middle_name.generic ?? []),
            ...(faker.definitions.person.middle_name.female ?? []),
          ];
          expect(female_applicable).toContain(name);

          name = faker.person.middleName('male');
          const male_applicable = [
            ...(faker.definitions.person.middle_name.generic ?? []),
            ...(faker.definitions.person.middle_name.male ?? []),
          ];
          expect(male_applicable).toContain(name);
        });

        it('should return a sex-specific middle name', () => {
          let name = fakerUK.person.middleName('female');
          expect(fakerUK.definitions.person.middle_name.female).toContain(name);

          name = fakerUK.person.middleName('male');
          expect(fakerUK.definitions.person.middle_name.male).toContain(name);
        });
      });

      describe('fullName()', () => {
        it('should return a name with firstName and lastName', () => {
          const fullName = faker.person.fullName();

          expect(fullName).toBeTypeOf('string');
          expect(fullName).toContain(' ');
        });

        it('should return a female applicable name without firstName and lastName', () => {
          const female_applicable = [
            ...(fakerMK.rawDefinitions.person?.prefix?.female ?? []),
            ...(fakerMK.rawDefinitions.person?.first_name?.female ?? []),
            ...(fakerMK.rawDefinitions.person?.last_name?.female ?? []),
            ...(fakerMK.rawDefinitions.person?.prefix?.generic ?? []),
            ...(fakerMK.rawDefinitions.person?.first_name?.generic ?? []),
            ...(fakerMK.rawDefinitions.person?.last_name?.generic ?? []),
            // ...(fakerMK.rawDefinitions.person?.suffix ?? []), Not applicable
          ];

          const fullName = fakerMK.person.fullName({ sex: 'female' });

          const parts = fullName.split(' ');
          for (const part of parts) {
            expect(female_applicable).toContain(part);
          }
        });

        it('should return a male applicable name without firstName and lastName', () => {
          const male_applicable = [
            ...(fakerMK.rawDefinitions.person?.prefix?.male ?? []),
            ...(fakerMK.rawDefinitions.person?.first_name?.male ?? []),
            ...(fakerMK.rawDefinitions.person?.last_name?.male ?? []),
            ...(fakerMK.rawDefinitions.person?.prefix?.generic ?? []),
            ...(fakerMK.rawDefinitions.person?.first_name?.generic ?? []),
            ...(fakerMK.rawDefinitions.person?.last_name?.generic ?? []),
            // ...(fakerMK.rawDefinitions.person?.suffix ?? []), Not applicable
          ];

          const fullName = fakerMK.person.fullName({ sex: 'male' });

          const parts = fullName.split(' ');
          for (const part of parts) {
            expect(male_applicable).toContain(part);
          }
        });

        it('should return a female applicable name with given firstName and lastName', () => {
          const female_applicable = [
            ...(fakerMK.rawDefinitions.person?.prefix?.female ?? []),
            ...(fakerMK.rawDefinitions.person?.prefix?.generic ?? []),
            'firstName',
            'lastName',
            // ...(fakerMK.rawDefinitions.person?.suffix ?? []), Not applicable
          ];

          const fullName = fakerMK.person.fullName({
            firstName: 'firstName',
            lastName: 'lastName',
            sex: 'female',
          });

          const parts = fullName.split(' ');
          for (const part of parts) {
            expect(female_applicable).toContain(part);
          }
        });

        it('should return a male applicable name with given firstName and lastName', () => {
          const male_applicable = [
            ...(fakerMK.rawDefinitions.person?.prefix?.male ?? []),
            ...(fakerMK.rawDefinitions.person?.prefix?.generic ?? []),
            'firstName',
            'lastName',
            // ...(fakerMK.rawDefinitions.person?.suffix ?? []), Not applicable
          ];

          const fullName = fakerMK.person.fullName({
            firstName: 'firstName',
            lastName: 'lastName',
            sex: 'male',
          });

          const parts = fullName.split(' ');
          for (const part of parts) {
            expect(male_applicable).toContain(part);
          }
        });
      });

      describe('gender()', () => {
        it('should return a default gender', () => {
          const gender = faker.person.gender();

          expect(gender).toBeTypeOf('string');
          expect(faker.definitions.person.gender).toContain(gender);
        });
      });

      describe('sex()', () => {
        it('should return a sex', () => {
          const sex = faker.person.sex();

          expect(sex).toBeTypeOf('string');
          expect(faker.definitions.person.sex).toContain(sex);
        });
      });

      describe('sexType()', () => {
        it('should return a sex type without generic by default', () => {
          const sexType = faker.person.sexType();

          expect(sexType).toBeTypeOf('string');
          expect([Sex.Female, Sex.Male]).toContain(sexType);
        });

        it('should return a sex type explicitly without generic', () => {
          const sexType = faker.person.sexType({ includeGeneric: false });

          expect(sexType).toBeTypeOf('string');
          expect([Sex.Female, Sex.Male]).toContain(sexType);
        });

        it('should return a sex type including generic', () => {
          const sexType = faker.person.sexType({ includeGeneric: true });

          expect(sexType).toBeTypeOf('string');
          expect(Object.values(Sex)).toContain(sexType);
        });
      });

      describe('prefix()', () => {
        it('should return a prefix', () => {
          const prefix = faker.person.prefix();

          expect(prefix).toBeTypeOf('string');
          const all_applicable = [
            ...(faker.definitions.person.prefix.generic ?? []),
            ...(faker.definitions.person.prefix.female ?? []),
            ...(faker.definitions.person.prefix.male ?? []),
          ];
          expect(all_applicable).toContain(prefix);
        });

        it('should return a generic prefix with given string', () => {
          const prefix = fakerMK.person.prefix('generic');

          expect(prefix).toBeTypeOf('string');
          expect(fakerMK.definitions.person.prefix.generic).toContain(prefix);
        });

        it('should return a female prefix with given string', () => {
          const prefix = fakerMK.person.prefix('female');

          expect(prefix).toBeTypeOf('string');
          const female_applicable = [
            ...(fakerMK.definitions.person.prefix.generic ?? []),
            ...(fakerMK.definitions.person.prefix.female ?? []),
          ];
          expect(female_applicable).toContain(prefix);
        });

        it('should return a male prefix with given string', () => {
          const prefix = fakerMK.person.prefix('male');

          expect(prefix).toBeTypeOf('string');
          const male_applicable = [
            ...(fakerMK.definitions.person.prefix.generic ?? []),
            ...(fakerMK.definitions.person.prefix.male ?? []),
          ];
          expect(male_applicable).toContain(prefix);
        });
      });

      describe('suffix()', () => {
        it('should return a suffix', () => {
          const suffix = faker.person.suffix();

          expect(suffix).toBeTypeOf('string');
          expect(faker.definitions.person.suffix).toContain(suffix);
        });
      });

      describe('jobTitle()', () => {
        it('should return a job title consisting of a descriptor, area, and type', () => {
          const jobTitle = faker.person.jobTitle();

          expect(jobTitle).toBeTypeOf('string');

          const [descriptor, level, job] = jobTitle.split(' ');

          expect(faker.definitions.person.job_descriptor).toContain(descriptor);
          expect(faker.definitions.person.job_area).toContain(level);
          expect(faker.definitions.person.job_type).toContain(job);
        });
      });

      describe('jobDescriptor()', () => {
        it('should return a descriptor', () => {
          const descriptor = faker.person.jobDescriptor();

          expect(descriptor).toBeTypeOf('string');

          expect(faker.definitions.person.job_descriptor).toContain(descriptor);
        });
      });

      describe('jobArea()', () => {
        it('should return a level', () => {
          const level = faker.person.jobArea();

          expect(level).toBeTypeOf('string');

          expect(faker.definitions.person.job_area).toContain(level);
        });
      });

      describe('jobType()', () => {
        it('should return a job', () => {
          const job = faker.person.jobType();

          expect(job).toBeTypeOf('string');

          expect(faker.definitions.person.job_type).toContain(job);
        });
      });

      describe('zodiacSign()', () => {
        it('returns a random zodiac sign', () => {
          const sign = faker.person.zodiacSign();

          expect(sign).toBeTypeOf('string');

          expect(faker.definitions.person.western_zodiac_sign).toContain(sign);
        });
      });
    }
  );
});
