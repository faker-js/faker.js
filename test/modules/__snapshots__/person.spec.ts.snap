// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`person > 42 > bio 1`] = `"traveler, philosopher, model"`;

exports[`person > 42 > firstName > noArgs 1`] = `"Moses"`;

exports[`person > 42 > firstName > with sex 1`] = `"Darnell"`;

exports[`person > 42 > fullName > noArgs 1`] = `"Moses Crist"`;

exports[`person > 42 > fullName > with all (sex) 1`] = `"John Doe"`;

exports[`person > 42 > fullName > with firstName 1`] = `"John Reynolds-Miller"`;

exports[`person > 42 > fullName > with lastName 1`] = `"Moses Doe"`;

exports[`person > 42 > fullName > with sex 1`] = `"Lorene Miller"`;

exports[`person > 42 > gender 1`] = `"Gender nonconforming"`;

exports[`person > 42 > jobArea 1`] = `"Identity"`;

exports[`person > 42 > jobDescriptor 1`] = `"National"`;

exports[`person > 42 > jobTitle 1`] = `"National Usability Producer"`;

exports[`person > 42 > jobType 1`] = `"Coordinator"`;

exports[`person > 42 > lastName > noArgs 1`] = `"Reynolds-Miller"`;

exports[`person > 42 > lastName > with sex 1`] = `"Wiegand"`;

exports[`person > 42 > middleName > noArgs 1`] = `"Quinn"`;

exports[`person > 42 > middleName > with sex 1`] = `"William"`;

exports[`person > 42 > prefix > noArgs 1`] = `"Mrs."`;

exports[`person > 42 > prefix > with sex 1`] = `"Dr."`;

exports[`person > 42 > sex > noArgs 1`] = `"female"`;

exports[`person > 42 > sex > with sex 1`] = `"female"`;

exports[`person > 42 > sexType 1`] = `"female"`;

exports[`person > 42 > suffix > noArgs 1`] = `"III"`;

exports[`person > 42 > suffix > with sex 1`] = `"III"`;

exports[`person > 42 > zodiacSign 1`] = `"Gemini"`;

exports[`person > 1211 > bio 1`] = `"decongestant supporter, parent ðŸŽ²"`;

exports[`person > 1211 > firstName > noArgs 1`] = `"Skye"`;

exports[`person > 1211 > firstName > with sex 1`] = `"Skye"`;

exports[`person > 1211 > fullName > noArgs 1`] = `"D'angelo Osinski-Paucek"`;

exports[`person > 1211 > fullName > with all (sex) 1`] = `"John Doe PhD"`;

exports[`person > 1211 > fullName > with firstName 1`] = `"Dr. John Fahey DDS"`;

exports[`person > 1211 > fullName > with lastName 1`] = `"Dr. D'angelo Doe DDS"`;

exports[`person > 1211 > fullName > with sex 1`] = `"Skye Zieme"`;

exports[`person > 1211 > gender 1`] = `"Trigender"`;

exports[`person > 1211 > jobArea 1`] = `"Factors"`;

exports[`person > 1211 > jobDescriptor 1`] = `"Chief"`;

exports[`person > 1211 > jobTitle 1`] = `"Chief Interactions Manager"`;

exports[`person > 1211 > jobType 1`] = `"Representative"`;

exports[`person > 1211 > lastName > noArgs 1`] = `"Fahey"`;

exports[`person > 1211 > lastName > with sex 1`] = `"Trantow"`;

exports[`person > 1211 > middleName > noArgs 1`] = `"Sage"`;

exports[`person > 1211 > middleName > with sex 1`] = `"Sage"`;

exports[`person > 1211 > prefix > noArgs 1`] = `"Ms."`;

exports[`person > 1211 > prefix > with sex 1`] = `"Ms."`;

exports[`person > 1211 > sex > noArgs 1`] = `"male"`;

exports[`person > 1211 > sex > with sex 1`] = `"male"`;

exports[`person > 1211 > sexType 1`] = `"male"`;

exports[`person > 1211 > suffix > noArgs 1`] = `"DVM"`;

exports[`person > 1211 > suffix > with sex 1`] = `"DVM"`;

exports[`person > 1211 > zodiacSign 1`] = `"Capricorn"`;

exports[`person > 1337 > bio 1`] = `"creator, engineer, friend"`;

exports[`person > 1337 > firstName > noArgs 1`] = `"Eva"`;

exports[`person > 1337 > firstName > with sex 1`] = `"Eugene"`;

exports[`person > 1337 > fullName > noArgs 1`] = `"Eva Hammes"`;

exports[`person > 1337 > fullName > with all (sex) 1`] = `"John Doe"`;

exports[`person > 1337 > fullName > with firstName 1`] = `"John Gottlieb"`;

exports[`person > 1337 > fullName > with lastName 1`] = `"Eva Doe"`;

exports[`person > 1337 > fullName > with sex 1`] = `"Marilyn Koelpin"`;

exports[`person > 1337 > gender 1`] = `"Demigender"`;

exports[`person > 1337 > jobArea 1`] = `"Functionality"`;

exports[`person > 1337 > jobDescriptor 1`] = `"Future"`;

exports[`person > 1337 > jobTitle 1`] = `"Future Marketing Engineer"`;

exports[`person > 1337 > jobType 1`] = `"Engineer"`;

exports[`person > 1337 > lastName > noArgs 1`] = `"Gottlieb"`;

exports[`person > 1337 > lastName > with sex 1`] = `"Cronin"`;

exports[`person > 1337 > middleName > noArgs 1`] = `"Dylan"`;

exports[`person > 1337 > middleName > with sex 1`] = `"Christopher"`;

exports[`person > 1337 > prefix > noArgs 1`] = `"Ms."`;

exports[`person > 1337 > prefix > with sex 1`] = `"Mr."`;

exports[`person > 1337 > sex > noArgs 1`] = `"female"`;

exports[`person > 1337 > sex > with sex 1`] = `"female"`;

exports[`person > 1337 > sexType 1`] = `"female"`;

exports[`person > 1337 > suffix > noArgs 1`] = `"I"`;

exports[`person > 1337 > suffix > with sex 1`] = `"I"`;

exports[`person > 1337 > zodiacSign 1`] = `"Taurus"`;
